### ISSUE TITLE
Editor freezes/crashes when inserting a reusable block twice

### ISSUE THREAD

**Issue by rfischmann** [WordPress/gutenberg#64514]
### Description

~When inserting a reusable block, converting it to a normal/editable block, and then trying to insert the same reusable block again, the editor freezes/crashes.~

The editor will freeze when a synced pattern (née reusable block) that contains a File block with _Show download button_ disabled.

### Step-by-step reproduction instructions

1. ~Insert a reusable block in a post.~
2. ~Convert the reusable block to a normal/editable block.~
3. ~Try to insert the same reusable block again.~

 

1. Create a new post.
2. Add a File block and upload a file.
3. Toggle _Show download button_ off.
5. Select File block, open Options menu, select _Create pattern_.
6. Name the pattern.
7. Editor will freeze/hang.

### Environment info

WordPress 6.6.1
Tried and was able to reproduced the issue on Firefox, Edge and Safari

### Please confirm that you have searched existing issues in the repo.

- [X] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [X] Yes

**Comment by Mamaduka** [comment]
Thanks for reporting the issue, @rfischmann!

I'm unable to reproduce the bug when testing with core blocks. Does this reusable block contain any custom blocks?

P.S. Could be related to #63909.

**Comment by skorasaurus** [comment]
I'm also unable to reproduce this using a pattern that contains a 3rd party block; this may be block-specific... could you share what block(s) you were using and copy and paste any message that appears?

**Comment by rdoughbs** [comment]
I was able to reproduce the freezing error with this block. It seems to freeze if you have a pattern with a file block with "show download button" toggled off. 

```
{
  "__file": "wp_block",
  "title": "bug",
  "content": "<!-- wp:paragraph -->\n<p>Add a file with \"show download button\" toggled off:</p>\n<!-- /wp:paragraph -->\n\n<!-- wp:file {\"id\":36,\"href\":\"http://localhost:8080/wp-content/uploads/2024/08/Test-11.pdf\",\"showDownloadButton\":false,\"displayPreview\":false} -->\n<div class=\"wp-block-file\"><a id=\"wp-block-file--media-fc02c5fa-5a65-49a3-a9d0-008b42375591\" href=\"http://localhost:8080/wp-content/uploads/2024/08/Test-11.pdf\">Test 11</a></div>\n<!-- /wp:file -->"
}
```

**Comment by noisysocks** [comment]
Synced patterns as a whole seem fine to me but can confirm that the editor hangs when creating or inserting a synced pattern that contains a File block with _Show download button_ disabled. Will update the title and description of this issue.

**Comment by Mamaduka** [comment]
The cause of the problem is the same as in #63909 - the attribute updates inside the effect. But it looks like the condition in the block is failing for some reason.

https://github.com/WordPress/gutenberg/blob/5109d5ba18b0fbb33ffd2c61c7793c16a5b31a58/packages/block-library/src/file/edit.js#L100-L105

**Comment by rfischmann** [comment]
Is it just me or this got even worse on 6.6.2? Now it's freezing/crashing browsers when we try to insert every new or edited reusable block to a post…

**Comment by Mamaduka** [comment]
@rfischmann, the fix (#65029) wasn't part of WP 6.6.2. Can you reproduce the same issue if you test with Gutenberg 19.2?

If so, please create a new issue with the reproduction steps. Please make sure to note every detail, like blocks inside the reusable block, etc.

**Comment by rfischmann** [comment]
> @rfischmann, the fix (#65029) wasn't part of WP 6.6.2. Can you reproduce the same issue if you test with Gutenberg 19.2?

Gutenberg 19.2 does fix it.

**Comment by Mamaduka** [comment]
Thank for confirming, @rfischmann!

I’ll mark this fix for consideration for the next minor release. Unfortunately, it wasn’t ready for this one.
